### DISCUSSION TITLE
"command": executable file not found in $PATH

### DISCUSSION THREAD

**Discussion by IvanWoo22** [nextflow-io/nextflow#4078] [Q&A]
### Description:
When I used nextflow on our LSF sys, I always get an error `Command 'ps' required by nextflow to collect task metrics cannot be found`. After checking the `.command.run` file and my singularity, I found it is not a `ps` PATH problem but a `command` PATH problem.
The following is my own testing process.
```bash
singularity exec --pid -B \
  nfmethylseq nfmethylseq/methylseq/workflow/../singularity-images/depot.galaxyproject.org-singularity-bismark-0.24.0--hdfd78af_0.img \
  command -v ps &>/dev/null || { >&2 echo "Command 'ps' required by nextflow to collect task metrics cannot be found"; }
  # Command 'ps' required by nextflow to collect task metrics cannot be found

singularity exec --pid -B \
  nfmethylseq nfmethylseq/methylseq/workflow/../singularity-images/depot.galaxyproject.org-singularity-bismark-0.24.0--hdfd78af_0.img \
  command -v ps
  # FATAL:   "command": executable file not found in $PATH

singularity exec --pid -B \
  nfmethylseq nfmethylseq/methylseq/workflow/../singularity-images/depot.galaxyproject.org-singularity-bismark-0.24.0--hdfd78af_0.img \
  ps
  # PID   USER     TIME  COMMAND
  #  1 wangq     0:00 sinit
  # 15 wangq     0:00 /bin/ps


```
When I comment the line `command -v ps &>/dev/null || { >&2 echo "Command 'ps' required by nextflow to collect task metrics cannot be found"; exit 1; }` with `#` and run with `bash .command.run`, it worked without error.

So, how could I fix this problem or no longer have this line of code every time the `.command.run` files generated?

**Comment by mribeirodantas** [comment]
Hey, Ivan. There is a list of tools in the [official documentation](https://www.nextflow.io/docs/latest/tracing.html?highlight=ps#tasks) that must be installed where tasks are run. `ps` is one of them. A solution to your problem is to have `ps` installed in your container 😄
